multimap/src/main/java/org/infinispan/multimap/impl/EmbeddedSetCache.java

```diff
@@ -36,43 +37,26 @@ public EmbeddedSetCache(Cache<K, SetBucket<V>> cache) {
    }
 
    /**
-    * Get the value as a collection
+    * Get the entry by key and return it as a set
     *
     * @param key, the name of the set
-    * @return the collection with values if such exist, or an empty collection if the key is not present
+    * @return the set with values if such exist, or null if the key is not present
     */
-   public CompletionStage<Collection<V>> get(K key) {
-      requireNonNull(key, ERR_KEY_CAN_T_BE_NULL);
-      return getEntry(key).thenApply(entry -> {
-         if (entry != null) {
-            return entry.getValue();
-         }
-         return Set.of();
-      });
-   }
-
-   private CompletionStage<CacheEntry<K, Collection<V>>> getEntry(K key) {
-      requireNonNull(key, ERR_KEY_CAN_T_BE_NULL);
-      return cache.getAdvancedCache().getCacheEntryAsync(key)
-            .thenApply(entry -> {
-               if (entry == null)
-                  return null;
-
-               return entryFactory.create(entry.getKey(),(entry.getValue().toSet()) , entry.getMetadata());
-            });
+   public CompletionStage<Set<V>> get(K key) {
+      return readWriteMap.eval(key, new SGetFunction<K, V>()).thenApply(v -> v != null ? v.values() : null);
```

Comment:
Katia might know more, but IIRC multimap has this thing of not returning null and instead an empty collection.

Reply:
changed in SGetFunction